src/V3LinkDot.cpp

```diff
@@ -224,8 +224,22 @@ class LinkDotState final {
 
     // METHODS
     static string nodeTextType(AstNode* nodep) {
-        if (VN_IS(nodep, Var)) {
-            return "variable";
+        if (const AstVar* const varp = VN_CAST(nodep, Var)) {
+            if (varp->isIO() || varp->isIfaceRef()) {
+                return "port";
+            } else if (varp->isGParam()) {
+                return "parameter";
+            } else if (varp->varType() == VVarType::LPARAM) {
+                return "local parameter";
+            } else {
+                return "variable";
+            }
+        } else if (const AstParamTypeDType* const dtypep = VN_CAST(nodep, ParamTypeDType)) {
+            if (dtypep->isGParam()) {
+                return "parameter";
+            } else {
+                return "local parameter";
```

Comment:
```suggestion
                return "type parameter";
            } else {
                return "local type parameter";
```

(Helps debug if user is trying to connect non-type to type)

Reply:
done